javaparser-core/src/main/java/com/github/javaparser/ast/type/ArrayType.java

```diff
@@ -118,6 +120,10 @@ public static Type wrapInArrayTypes(Type type, List<ArrayBracketPair>... arrayBr
                 }
             }
         }
+        // The above code concatenates the arrayBracketPairLists, because they represent different
+        // parts of the array rather than two sources of information about the same parts of the
+        // array.
+        type.setArrayBracketPairs(arrayAnnotations);
```

Comment:
The array bracket pairs were removed from the API because they were error prone and probably-usually not the best way to deal with arrays. That's why we introduced the wrapping `ArrayType`. That one has annotations for every bracket level. What is the reason you're adding the bracket pairs back to the API?